netbox_onboarding/onboard.py

```diff
@@ -422,16 +422,28 @@ def ensure_device_role(
         self.netdev.ot.save()
         return
 
-    def ensure_device_instance(self):
-        """Ensure that the device instance exists in NetBox and is assigned the provided device role or DEFAULT_ROLE."""
-        # TODO: this can create duplicate entries in NetBox...
-        device, _ = Device.objects.get_or_create(
-            name=self.netdev.hostname,
-            device_type=self.device_type,
-            device_role=self.netdev.ot.role,
-            platform=self.netdev.ot.platform,
-            site=self.netdev.ot.site,
-        )
+    def ensure_device_instance(self, default_status=PLUGIN_SETTINGS["default_device_status"]):
+        """Ensure that the device instance exists in NetBox and is assigned the provided device role or DEFAULT_ROLE.
+
+        Args:
+          default_status (str) : status assigned to a new device by default.
+        """
+        try:
+            device = Device.objects.get(
+                name=self.netdev.hostname,
+                device_type=self.device_type,
+                device_role=self.netdev.ot.role,
```

Comment:
Would it make sense to move `device_type` and `device_role` out of the `get()` call (so we match the device based only on name and site), and possibly update them even if we have an existing device? I'm thinking something like:

```python
try:
    device = Device.objects.get(name=self.netdev.hostname, site=self.netdev.ot.site)
except Device.DoesNotExist:
    device = Device.objects.create(name=self.netdev.hostname, site=self.netdev.ot.site, status=default_status)

device.device_type = self.device_type
device.device_role = self.netdev.ot.role
device.platform = self.netdev.ot.platform
device.serial = self.netdev.serial_number
device.save()
```

Reply:
@glennmatthews thanks, I think it's a great idea
I remove `device_role` and `device_type` from the get() but I couldn't remove them from `create()` because both are mandatory.
Now if a device already exist but doesn't have a platform, it will be updated as part of the onboarding as well as the serial number.